### PR TITLE
update NPM dep versions

### PR DESCRIPTION
**Pull request by jessp01**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,20 +25,20 @@
...
     "handlebars": "^4.0.11",
     "normalize-package-data": "^2.4.0",
     "npm-remote-ls": "~1.3.2",
-    "request": "~2.81.0",
-    "semver": "^5.4.1",
-    "tar": "~2.2.1",
-    "tmp": "~0.0.28"
+    "request": "~2.87.0",
+    "semver": "^5.5.0",
+    "tar": "~4.4.4",
```
Comment (ekohl) [inline package.json]: It looks like the API changed:
```
$ ./bin/npm2rpm.js -n patternfly-react -s single -v 2.10.1
---- npm2rpm ----
-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
 - Starting npm module download: https://registry.npmjs.org/patternfly-react/-/patternfly-react-2.10.1.tgz
 - Unpacking in /tmp/npm2rpm-26409vV0eZwLIIro4 ...
/home/ekohl/dev/npm2rpm/lib/npm_helpers.js:21
  var write_stream = tar.Extract({path: tmpDir, strip: 0})
                         ^

TypeError: tar.Extract is not a function
    at Object.module.exports.extractTar (/home/ekohl/dev/npm2rpm/lib/npm_helpers.js:21:26)
    at Object.<anonymous> (/home/ekohl/dev/npm2rpm/bin/npm2rpm.js:40:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
```

File: package.json [hunk 2]
```diff
@@ -25,20 +25,20 @@
...
     "datejs": "*",
     "handlebars": "^4.0.11",
     "normalize-package-data": "^2.4.0",
     "npm-remote-ls": "~1.3.2",
-    "request": "~2.81.0",
-    "semver": "^5.4.1",
-    "tar": "~2.2.1",
-    "tmp": "~0.0.28"
+    "request": "~2.87.0",
+    "semver": "^5.5.0",
```
Comment (ekohl) [inline package.json]: This change isn't needed. We work with `^5.4.1` and that allows the same max versions as `^5.5.0`.

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by jessp01** [comment]
Hi @ekohl,

You're absolutely right about the `tar` NPM. I failed to re-test after upgrading that one to the latest version, my apologies. I reverted the version and will submit a separate pull to support the newer version.
I also reverted the semver module to an earlier version as you requested but can you please explain why you object to upgrading to latest?

Really, the most important thing about this pull is the upgrade of the `request` module as it depends on the `hoek` module for which an exploit was recently reported and fixed:
https://nvd.nist.gov/vuln/detail/CVE-2018-3728

If I'm honest, that's not critical in the case for something like `npm2rpm` but still best to upgrade:)

**Comment by ekohl** [comment]
@jessp01 generally I prefer open dependencies. I think there are a few more where we now use `~` instead of `^`. That would allow using a much wider range of versions including those with fixed vulnerabilities. If they properly follow semver then it should still work API-wise while allowing the security updates. An example of this is our semver dependency: it allows installing the latest 5.x but we're very wide in which versions we allow.

**Comment by ekohl** [comment]
Could you also add `package-lock.json` to `.gitignore` btw?

**Comment by jessp01** [comment]
@ekohl, semver - understood. package-lock.json added to .gitignore.

**Comment by ekohl** [comment]
Fixed in https://github.com/dLobatog/npm2rpm/pull/48

**Pull request closed without merging**